bodhi/server/services/releases.py

```diff
@@ -325,6 +325,15 @@ def save_release(request):
             r = request.db.query(Release).filter(Release.name == edited).one()
             for k, v in data.items():
                 setattr(r, k, v)
+                # We have to change status for release updates to obsolete
+                if k == "state" and v == ReleaseState.archived:
+                    updates = request.db.query(Update).filter(Update.release_id == r.id).filter(
+                            and_(*[Update.status != s for s in (
+                                UpdateStatus.obsolete, UpdateStatus.stable
```

Comment:
We should also include unpushed.

You can express the above in a simpler fashion:

```
updates = request.db.query(Update).filter(Update.release_id == r.id).filter(
    Update.status.notin_([UpdateStatus.obsolete, UpdateStatus.stable, UpdateStatus.unpushed])
```